Add vm_deploy & vm_destroy scripts.

jasondoze opened the pull request:
Add go_install.sh.

Add README.md.

Add .gitignore.

jasondoze committed:
Add vm_deploy & vm_destroy scripts.

Add go_install.sh.

Add README.md.

Add .gitignore.

Signed-off-by: jason <dozejason@gmail.com>

jasondoze committed:
Add exclude files to rsync transfer.

Comment out apt-cache show golang, its in the apt cache but appears to still be updated for go install.

Modify go_install.sh conditionals to initialize and run cobra-cli.

Signed-off-by: jason <dozejason@gmail.com>

jasondoze committed:
Add cobra install, cobra-cli init will not setup all the framework that cobra init does.

Add exclude files to rsync tranfer.

Signed-off-by: jason <dozejason@gmail.com>

jasondoze committed:
Add go and python directories.

Add python_install.sh script.

Add ssh execute for python script to vm_deploy.sh.

Signed-off-by: jason <dozejason@gmail.com>

stephenlauck commented on python/python_install.sh:
```diff
@@ -0,0 +1,57 @@
+  python3 -m venv py3env
+fi
+
+# Activate virtual environment
```
How would you check this?

jasondoze replied:
Not sure how to handle activating the environment. This really doesnt work bc once I ssh into the vm, the environment is no longer activated. What direction would you suggest?

stephenlauck replied:
I suggest you learn about user shell context and login context. If you are "sourcing and env" what does that mean? What is the context of a shell? What is passed to a subshell? Have you thought about if environment variables work in subshells? What is the scope?

stephenlauck replied:
What is the difference between the .bashrc, .profile in your home directory (if you have those files)?

stephenlauck commented on vm_deploy.sh:
```diff
@@ -76,7 +76,8 @@ echo -e "\n\033[1;32m==== Transferring files to VM ====\033[0m\n"
 echo -e "\n\033[1;32m==== Execute install scripts on VM ====\033[0m\n"
-ssh -o StrictHostKeyChecking=no -i ./id_ed25519 $USER@$(multipass info nameless | grep IPv4 | awk '{ print $2 }') 'cd meshellica && bash go_install.sh'
+ssh -o StrictHostKeyChecking=no -i ./id_ed25519 $USER@$(multipass info nameless | grep IPv4 | awk '{ print $2 }') 'cd meshellica/go && bash go_install.sh'
+ssh -o StrictHostKeyChecking=no -i ./id_ed25519 $USER@$(multipass info nameless | grep IPv4 | awk '{ print $2 }') 'cd meshellica/python && bash python_install.sh'
```
is there a way to pass something to this script to decide which line to run?

jasondoze replied:
Like what we did yesterday?

```
install.sh
if [ "$1" == "go" ]
then
    ssh -o StrictHostKeyChecking=no -i ./id_ed25519 $USER@$(multipass info nameless | grep IPv4 | awk '{ print $2 }') 'cd meshellica/go && bash go_install.sh'
elif ......

bash install.sh go
```

stephenlauck replied:
What about just one line with variables passed?

`cd meshellica/$LANG && bash $LANG_install.sh`

jasondoze committed:
Add conditional to pass arguement to vm_deploy script ro run go or python install.

Thats dope! Didnt know arguements could be passed to running scripts.

Signed-off-by: jason <dozejason@gmail.com>

jasondoze committed:
Add case conditional to define arguments for running go & python scripts.

Remove apt-cache for python3, its virtual so need another solution.

Add virtual environment to .profile, it works but in all directories.

Signed-off-by: jason <dozejason@gmail.com>

jasondoze committed:
Add $PROGRAM to go case command to run go scripts.

Signed-off-by: jason <dozejason@gmail.com>

stephenlauck commented on vm_deploy.sh:
```diff
@@ -75,23 +75,21 @@ fi
+    ;;
+  "python")
+    echo -e "\n\033[1;32m==== Running $PROGRAM scripts ====\033[0m\n"
+    ssh -o StrictHostKeyChecking=no -i ./id_ed25519 $USER@$(multipass info nameless | grep IPv4 | awk '{ print $2 }') "cd meshellica/$PROGRAM && bash install_$PROGRAM.sh"
```
You don't even need the case statement if the $PROGRAM is in the same place in all the commands and the rest of the command is the same.
